guidance/library/_json.py

```diff
@@ -718,10 +703,182 @@ def anyOf(
         return lm + select(options)
 
+    @guidance(stateless=True)
+    def oneOf(
+        self,
+        lm,
+        *,
+        oneof_list: Sequence[JSONSchema],
+        base_uri: str,
+    ):
+        if len(oneof_list) == 1:
+            return lm + self.json(json_schema=oneof_list[0], base_uri=base_uri)
+        warnings.warn("oneOf not fully supported, falling back to anyOf. This may cause validation errors in some cases.")
+        return lm + self.anyOf(anyof_list=oneof_list, base_uri=base_uri)
+
+    @guidance(stateless=True)
+    def allOf(
+        self,
+        lm,
+        *,
+        parent_schema: JSONSchema,
+        base_uri: str,
+    ):
+        type = set(JSONType)
+        properties: defaultdict[str, list[JSONSchema]] = defaultdict(list)
+        required: set[str] = set()
+        additional_properties_list: list[JSONSchema] = []
+        items_list: list[JSONSchema] = []
+        other_data: dict[str, Any] = {}
+        enum: Optional[list[Any]] = None
+        const: Union[Unset, Any] = _unset
+
+        def handle_keyword(key: str, value: Any, base_uri: str):
+            nonlocal type
+            nonlocal required
+            nonlocal const
+            nonlocal enum
+
+            if key == Keyword.REF:
+                ref = cast(str, value)
+                abspath = urijoin(base_uri, ref)
+                resolved = self._resolver.lookup(abspath)
+                add_schema(resolved.contents, base_uri=resolved.resolver._base_uri)
+
+            elif key == Keyword.CONST:
+                if const is not _unset and const != value:
+                    raise ValueError(f"allOf with multiple conflicting const values: {const!r} and {value!r}")
+                const = value
+
+            elif key == Keyword.ENUM:
+                value = cast(list[Any], value)
+                if enum is not None:
+                    try:
+                        enum = list(set(enum) & set(value))
+                    except TypeError:
+                        # Check on equality, not on hash
+                        # Yes, this is O(n^2).
+                        # Hope the items were unique.
+                        # ¯\_(ツ)_/¯
+                        enum = [a for a in enum for b in value if a == b]
+                else:
+                    enum = value
+
+            elif key == Keyword.TYPE:
+                value = cast(Union[str, list[str]], value)
+                if isinstance(value, str):
+                    value_set = {value}
+                else:
+                    value_set = set(value)
+                if JSONType.NUMBER in value_set:
+                    # Number implies integer
+                    value_set.add(JSONType.INTEGER)
+                type &= value_set
+                # Throw an error early if we have conflicting types
+                if not type:
+                    raise ValueError("allOf with conflicting types")
```

Comment:
Can we tell users what the conflict was? Otherwise you're telling the user "I didn't find something, but I'm not telling you what I was looking for or where I was looking for it."

Reply:
Maybe I can defer the actual intersection logic to outside the loop here so I can show the offending types directly. Good call.